invenio_drafts_resources/records/api.py

```diff
@@ -33,57 +55,66 @@ class ParentRecord(RecordBase):
 
 
 class Record(RecordBase):
-    """Record base API.
+    """Record API."""
 
-    Note: This class is meant to work along a with a draft class.
-    """
+    #
+    # Configuration to be set by a subclass
+    #
 
-    # Configuration
+    #: The record's SQLAlchemy model class. Must be set by the subclass.
     model_cls = None
-
+    #: The parent state's SQLAlchemy model class. Must be set by the subclass.
+    versions_model_cls = None
+    #: The parent record's API class. Must be set by the subclass.
+    parent_record_cls = None
+
+    #
+    # System fields
+    #
+    #: The internal persistent identifier. Records and drafts share UUID.
     pid = PIDField('id', provider=DraftRecordIdProviderV2, delete=True)
 
+    #: System field to check if a record has been published.
     is_published = PIDStatusCheckField(status=PIDStatus.REGISTERED)
 
-    #: List of field names (strings) to copy from the draft on create.
-    create_from_draft = []
+    #: The parent record - the draft is responsible for creating the parent.
+    parent = ParentField(
+        ParentRecord, create=False, soft_delete=False, hard_delete=False)
```

Comment:
The value is just used in a single place to do a ``isinstance(val, ParentRecord)`` so in this case it's ok as long as the base classes that's passed in. Its because I'm relying on ``RelatedModelField`` and this value is not that easy to overwrite.

Reply:
If setting `parent_record_cls` doesn't affect the `ParentField` then, should we just not pass `ParentRecord` as an argument to `ParentField` ? ... we can discuss this one